server/src/main/java/org/apache/druid/server/coordination/BatchDataSegmentAnnouncer.java

```diff
@@ -99,13 +107,28 @@ public BatchDataSegmentAnnouncer(
       return rv;
     };
 
-    if (this.config.isSkipSegmentAnnouncementOnZk()) {
+    isSkipSegmentAnnouncementOnZk = !zkEnablementConfig.isEnabled() || config.isSkipSegmentAnnouncementOnZk();
```

Comment:
It seems like this class is almost entirely to handle zk stuff, does it need to be bound and exist at all if zk is disabled?

Reply:
this is also calling methods in
`private final ChangeRequestHistory<DataSegmentChangeRequest> changes = new ChangeRequestHistory<>();`

which is used by the http endpoint for segment sync.

However, yeah it takes more work to announce stuff in zk so  it looks like that this class is primarily doing that whereas it is actually supporting both.
If/When we are able to delete zk based segment management this class would shrink significantly.
Also, unrelated, it could possibly be refactored to separate two things it is doing so that things are more clear. 
However, you know my preference, just delete  zk stuff :)